Sources/UI/Camera/PhotoCamera.swift

```diff
@@ -68,16 +60,32 @@ public extension PhotoCamera {
       self.photoOutput.capturePhoto(with: photoSettings, delegate: self)
     }
   }
+  
+  func takePhoto(with photoSettings: AVCapturePhotoSettings, videoOrientation: AVCaptureVideoOrientation? = nil) {
+    let videoOrientation = (videoOrientation != nil) ? videoOrientation : previewLayer.connection?.videoOrientation
+    sessionQueue.async {
+      guard self.session.isRunning else {
```

Comment:
Do we have a retain cycle here?

Reply:
I ran this test on a real device and there doesn't seem to be a retain cycle
```
func test_sut_isDeallocated() async {
  var sut: PhotoCamera? = PhotoCamera()
  
  try? await sut!.prepare()
  sut!.startSession()
  sut!.takePhoto()
  
  sut = nil
  
  XCTAssertNil(sut)
}
  ```
  
  The test passed and also `Camera` was deinitialized

Reply:
There should be no problem with retaining because the queue will, at some point, execute the work item and remove a reference. But I don't understand the test here; you are setting a value of `Optional<PhotoCamera>` to `.none` and then checking if it is `.none`. Of course, this will always be the case. Or am I missing something?

Reply:
Looks like there is a retain cycle after running this:
```
func expectNoMemoryLeaks(for object: AnyObject, file: StaticString = #filePath, line: UInt = #line) {
  addTeardownBlock { [weak object] in
   XCTAssertNil(object, file: file, line: line)
  }
}
  ```